doc/man/lttng-ust.3.txt

```diff
@@ -0,0 +1,1030 @@
+application: it must be loaded manually. This is why the application is
+built with no mention of this tracepoint provider, but still needs
+libdl:
+
+[role="term"]
+--------------------------------
+cc -o app app.o tp-define.o -ldl
+--------------------------------
+
+There are two ways to dynamically load the tracepoint provider shared
+object:
+
+  * Load it manually from the application using man:dlopen(3)
+  * Make the dynamic loader load it with the `LD_PRELOAD`
+    environment variable (see man:ld.so(8))
+
+If the application does not dynamically load the tracepoint provider
+shared object using one of the methods above, tracing is disabled for
+this application, and the events are not listed in the output of
+man:lttng-list(1).
+
+Note that it is not safe to use man:dlclose(3) on a tracepoint provider
+shared object that is being actively used for tracing, due to a lack of
+reference counting from LTTng-UST to the shared object.
+
+For example, statically linking a tracepoint provider to a shared object
+which is to be dynamically loaded by an application (a plugin, for
+example) is not safe: the shared object, which contains the tracepoint
+provider, could be dynamically closed (man:dlclose(3)) at any time by
+the application.
+
+To instrument a shared object, either:
+
+  * Statically link the tracepoint provider to the application, or
+  * Build the tracepoint provider as a shared object (following the
+    procedure shown in this section), and preload it when tracing is
+    needed using the `LD_PRELOAD` environment variable.
+
+
+Using LTTng-UST with daemons
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Some extra care is needed when using `liblttng-ust` with daemon
+applications that call man:fork(2), man:clone(2), or BSD's man:rfork(2)
+without a following man:exec(3) family system call. The library
+`liblttng-ust-fork.so` needs to be preloaded before starting the
+application with the `LD_PRELOAD` environment variable (see
+man:ld.so(8)).
+
+
+Context information
+~~~~~~~~~~~~~~~~~~~
+Context information can be prepended by the LTTng-UST tracer before
+each event, or before specific events. The following context fields
+are supported by LTTng-UST:
+
+`vtid`::
+    Virtual thread ID: thread ID as seen from the point of view of
+    the process namespace.
+
+`vpid`::
+    Virtual process ID: process ID as seen from the point of view of
+    the process namespace.
+
+`ip`::
+    Instruction pointer: enables recording the exact address from which
+    an event was emitted. This context field can be used to
+    reverse-lookup the source location that caused the event
+    to be emitted.
+
+`procname`::
+    Thread name, as set by man:exec(3) or man:prctl(2). It is
+    recommended that programs set their thread name with man:prctl(2)
+    before hitting the first tracepoint for that thread.
+
+`pthread_id`::
+    POSIX thread identifier. Can be used on architectures where
+    `pthread_t` maps nicely to an `unsigned long` type.
+
```

Comment:
make sure to try the cpu_id context to ensure it can be used as an event context in lttng-tools (unsure).